primitives/message-lane/src/lib.rs

```diff
@@ -158,11 +158,36 @@ impl Default for OutboundLaneData {
 }
 
 /// Returns total number of messages in the `InboundLaneData::relayers` vector.
+///
+/// Returns `None` if there are more messages that `MessageNonce` may fit (i.e. `MessageNonce + 1`).
 pub fn total_unrewarded_messages<RelayerId>(
 	relayers: &VecDeque<(MessageNonce, MessageNonce, RelayerId)>,
-) -> MessageNonce {
+) -> Option<MessageNonce> {
 	match (relayers.front(), relayers.back()) {
-		(Some((begin, _, _)), Some((_, end, _))) => end.checked_sub(*begin).and_then(|d| d.checked_add(1)).unwrap_or(0),
-		_ => 0,
+		(Some((begin, _, _)), Some((_, end, _))) => {
+			if let Some(difference) = end.checked_sub(*begin) {
+				difference.checked_add(1)
```

Comment:
1. The reason is that `(begin; end)` pair means inclusive range here. And to specify empty inclusive range you shall make `end` lesser than `begin` - e.g. `10..=9`. So this is never an error to a caller.
2. `saturated_add` would return wrong result, just dumping error here without any notifications to the caller.If this function will ever be called from the place where correct result is important, this may lead to unexpected sensitive errors. Both existing calls of this function are `unwrap_or()`-ing result because it makes sense to do that there. First call - because `u64::MAX` may not be valid value of `relayers_state.total_messages` anyway. Second call just return stats && isn't used in sensitive context.